core/src/test/java/io/crate/types/DataTypesTest.java

```diff
@@ -240,4 +240,118 @@ public void testConvertBooleanToShort() {
     public void testConvertBooleanToByte() {
         DataTypes.BYTE.value(true);
     }
+
+    @Test
+    public void testLongTypeCompareValueToWith() {
+        Long val1 = DataTypes.LONG.value(null);
+        Long val2 = DataTypes.LONG.value(null);
+        assertThat(LongType.INSTANCE.compareValueTo(val1, val2), is(0));
+
+        val1 = DataTypes.LONG.value(null);
+        val2 = DataTypes.LONG.value(2L);
+        assertThat(LongType.INSTANCE.compareValueTo(val1, val2), is(-1));
+
+        val1 = DataTypes.LONG.value(2L);
+        val2 = DataTypes.LONG.value(2L);
+        assertThat(LongType.INSTANCE.compareValueTo(val1, val2), is(0));
+
+        val1 = DataTypes.LONG.value(1L);
+        val2 = DataTypes.LONG.value(null);
+        assertThat(LongType.INSTANCE.compareValueTo(val1, val2), is(1));
+    }
+
+    @Test
+    public void testIntTypeCompareValueTo() {
+        Integer val1 = DataTypes.INTEGER.value(null);
+        Integer val2 = DataTypes.INTEGER.value(null);
+        assertThat(IntegerType.INSTANCE.compareValueTo(val1, val2), is(0));
+
+        val1 = DataTypes.INTEGER.value(null);
+        val2 = DataTypes.INTEGER.value(2);
+        assertThat(IntegerType.INSTANCE.compareValueTo(val1, val2), is(-1));
+
+        val1 = DataTypes.INTEGER.value(2);
+        val2 = DataTypes.INTEGER.value(2);
+        assertThat(IntegerType.INSTANCE.compareValueTo(val1, val2), is(0));
+
+        val1 = DataTypes.INTEGER.value(1);
+        val2 = DataTypes.INTEGER.value(null);
+        assertThat(IntegerType.INSTANCE.compareValueTo(val1, val2), is(1));
+    }
+
+    @Test
+    public void testDoubleTypeCompareValueTo() {
+        Double val1 = DataTypes.DOUBLE.value(null);
+        Double val2 = DataTypes.DOUBLE.value(null);
+        assertThat(DoubleType.INSTANCE.compareValueTo(val1, val2), is(0));
+
+        val1 = DataTypes.DOUBLE.value(null);
```

Comment:
I think it would make sense to create a helper function where you can pass the type, the two values and the expected result